### PR TITLE
PE-2814: GQLEdges to SnapshotData Stream Transform

### PR DESCRIPTION
**Pull request by matibat**
Changes
- Implements `gqlEdgesToSnapshotDataStreamTransform` which is a stream transform: takes Stream<TxSnapshot> and maps to Stream<UInt8List> containing the raw JSON data of SnapshotData.

This method is gonna be used by SnapshotItemToBeCreated to construct the Transaction Data Stream.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/21buk3o0h8oa0
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/7b1nfag4u8t1g

### CODE REVIEW

File: lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:ardrive/utils/snapshots/snapshot_types.dart';
+
+// Function that maps a stream of TxSnapshot to a stream of JSON serialized objects as Uint8Array
+Stream<Uint8List> gqlEdgesToSnapshotDataStreamTransform(
```
Comment (matibat) [inline lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart:7]: I think `UintList` is the correct type for streaming bytes to the network, right? We don't have to stream individual bytes.

File: lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  // Yield the beginning of a JSON array
+  yield Uint8List.fromList(utf8.encode('{"txSnapshots":['));
+
+  // Set a flag to determine if this is the first object in the stream
+  var first = true;
+
+  // Iterate through the stream
+  await for (final txSnapshot in stream) {
+    // If this is not the first object in the stream, yield a comma separator
+    if (!first) {
```
[the remark this answers is not in the document]
Comment (matibat) [reply lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart:21]: I'm actually using that flag to determine if I have to put a comma before the item in the array (in the input stream actually). I could use instead an index and yield the `','` when `index != 0`.

[the remark this answers is not in the document]
Comment (matibat) [reply lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart]: Applied - Also made a class non-abstract.
5262d92efaa86832cf3718d7ce7a8a9257bf4a43

[the remark this answers is not in the document]
Comment (matibat) [reply lib/utils/snapshots/gql_edges_to_snapshot_data_stream_transform.dart]: Forgot to commit the tests 🤡 
[fb40542](https://github.com/ardriveapp/ardrive-web/pull/845/commits/fb40542e7e58aa4d88a15c23ac1b8b5a50b7eaac)

File: test/utils/gql_edges_to_snapshot_data_stream_transform_test.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          },
+          'tags': [],
+        },
+      ),
+      jsonMetadata: '{"name": "name_$i"}',
+    );
+  }
+}
+
+// Test sheet for the gqlEdgesToSnapshotDataStreamTransform method
```
Comment (matibat) [inline test/utils/gql_edges_to_snapshot_data_stream_transform_test.dart:36]:
```suggestion
```

File: lib/utils/snapshots/tx_snapshot_to_snapshot_data.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+// StreamTransformer that converts a stream of TxSnapshot objects into a stream of Uint8List objects
+StreamTransformer<TxSnapshot, Uint8List> txSnapshotToSnapshotData =
+    StreamTransformer.fromBind((stream) async* {
+  // Use a JSON encoder to serialize the objects in the stream.
+  const encoder = JsonEncoder();
+
+  // Yield the beginning of a JSON array
+  yield Uint8List.fromList(utf8.encode('{"txSnapshots":['));
+
+  int index = 0;
```
[the remark this answers is not in the document]
Comment (matibat) [reply lib/utils/snapshots/tx_snapshot_to_snapshot_data.dart]: Back to boolean, but with a more descriptive name: 4234240ddd0a658a9bd4fd77364eeddfbdbb5592

**Review by karlprieb — APPROVED** [review]

**Pull request merged**